core/src/apps/bitcoin/keychain.py

```diff
@@ -150,6 +150,8 @@ def validate_path_against_script_type(
             append(PATTERN_GREENADDRESS_B)
         if coin.coin_name in BITCOIN_NAMES:
             append(PATTERN_CASA_UNHARDENED)
+            append(PATTERN_UNCHAINED_HARDENED)
+            append(PATTERN_UNCHAINED_UNHARDENED)
```

Comment:
> Unchained plans to support p2wsh [...]

You are adding the paths under `SPENDP2SHWITNESS`. That's "legacy SegWit", i.e. [P2WPKH](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#p2wpkh-nested-in-bip16-p2sh) or [P2WSH](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#p2wsh-nested-in-bip16-p2sh) nested in BIP 16 P2SH. If you want to use the paths with P2WSH, then you should place them under `SPENDWITNESS`.

Reply:
change in 62caf4c12f6eb822fbe7b24422c95b972a483301